{{cookiecutter.repo_name}}/README.rst

```diff
@@ -12,12 +12,118 @@ The XBlock SDK Workbench, including this XBlock, will be available on the list o
 Translating
 -----------
 
+Internationalization (i18n) is when a program is made aware of multiple languages.
+Localization (l10n) is adapting a program to local language and cultural habits.
+
 Use the locale directory to provide internationalized strings for your XBlock project.
+For more information on how to enable translations, visit the
+[Open edX XBlock tutorial on Internationalization](http://edx.readthedocs.org/projects/xblock-tutorial/en/latest/edx_platform/edx_lms.html).
+
+This cookiecutter template uses [django-statici18n](https://django-statici18n.readthedocs.io/en/latest/)
+to provide translations to static javascript using `gettext`.
+
+The included Makefile contains targets for extracting, compiling and validating translatable strings.
+The general steps to provide multilingual messages for a Python program (or an XBlock) are:
+
+1. Mark translatable strings.
+2. Run i18n tools to create raw message catalogs.
+3. Create language specific translations for each message in the catalogs.
+4. Use `gettext` to translate strings.
+
+### 1. Mark translatable strings
+
+Mark translatable strings in python:
+
+```python
+from django.utils.translation import ugettext as _
+
+# Translators: This comment will appear in the `.po` file.
+message = _("This will be marked.")
+```
+
+See [edx-developer-guide](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/internationalization/i18n.html#python-source-code)
+for more information.
+
+You can also use `gettext` to mark strings in javascript:
+
+```javascript
+// Translators: This comment will appear in the `.po` file.
+var message = gettext("Custom message.");
+```
+
+See [edx-developer-guide](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/internationalization/i18n.html#javascript-files)
+for more information.
+
+### 2. Run i18n tools to create Raw message catalogs
+
+This cookiecutter template offers multiple make targets which are shortcuts to
+use [edx-i18n-tools](https://github.com/edx/i18n-tools).
+
+After marking strings as translatable we have to create the raw message catalogs.
+These catalogs are created in `.po` files. For more information see
+[GNU PO file documentation](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html).
+These catalogs can be created by running:
+
+```shell
+$ make extract_translations
+```
+
+The previous command will create the necessary `.po` files under
+`{{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/locale/en/LC_MESSAGES/text.po`.
+The `text.po` file is created from the `django-partial.po` file created by
+[`django-admin makemessages`](https://docs.djangoproject.com/en/1.11/topics/i18n/translation/#message-files),
+this is why you will not see a `django-partial.po` file.
+
+### 3. Create language specific translations
+
+After creating the raw message catalogs, all translations should be filled out by the translator.
```

Comment:
I believe translating and compile should be independent steps here.

There are transifex make targets to push / pull translations and would be nice to have some description of this process as well.

Reply:
Good point, maybe a substep? That way moving the transifex instructions here look better, too.

Reply:
Updated this to add 3.1 and 3.2. Also pointed to the transifex documentation about Django.

Reply:
Alright 👍